mingw/MSVC patches

vladimirgamalyan opened the pull request:
Default installation from www.mingw.org has [known bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52015) (std::to_string does not work under MinGW).
I suggest another way to build the string via std::stringstream (it could be little slower).
Also, MSVC has own view on expand VA_ARGS (similar issue on [stackoverlow](http://stackoverflow.com/questions/5134523/msvc-doesnt-expand-va-args-correctly)). There is a little patch for that case (tester_ methods placed directy into EXPECT_EXCEPTION, EXPECT_TRUE and EXPECT_EQUAL macros). I'm not sure about that "prepromagic", but test.cc prints same results: [Travis CI](https://travis-ci.org/vladimirgamalian/testing.h/jobs/73878130) and [AppVeyour](https://ci.appveyor.com/project/vladimirgamalian/testing-h/build/job/m8w6sc3uc7i5goa2).

AMDmi3 commented:
These changes look good, however please split/merge them into a set of complete topic commits:
- readme fix
- float constant fix
- typo fixes
- disabling color on win32
- dropping std::to_string
- fixing unused catch argument
- adding wrapper macros for win32

METHOD_WRAPPER_LAMBDA may be removed in a separate commit, as it's not used anywhere

Also please indent nested preprocessor statements with tabs, e.g.

```
#if FOO
#   define ...
#endif
```

vladimirgamalyan committed:
Fix README.md

EXPECT_STRING macro has been removed from the example.

vladimirgamalyan committed:
Fix float constant

vladimirgamalyan committed:
Fix typos

vladimirgamalyan committed:
Drop std::to_string

According gcc bug 52015: std::to_string does not work under MinGW.
Presently, the default installation from www.mingw.org has it.

vladimirgamalyan committed:
Remove unused catch argument

vladimirgamalyan committed:
Add wrapper macros for win32

MSVS doesn't expand __VA_ARGS__ some way as GCC.

vladimirgamalyan committed:
Disable color on win32

vladimirgamalyan committed:
Remove unused macro

vladimirgamalyan commented:
Done.

AMDmi3 commented:
Perfect, thanks!
